operator/k8s_node.go

```diff
@@ -120,12 +120,26 @@ func runNodeWatcher() error {
 
 	go func() {
 		cache.WaitForCacheSync(wait.NeverStop, nodeController.HasSynced)
+
 		serNodes.Enqueue(func() error {
 			// Since we serialize all events received from k8s we know that
 			// at this point the list in k8sNodeStore should be the source of truth
 			// and we need to delete all nodes in the kvNodeStore that are *not*
 			// present in the k8sNodeStore.
 
+			if enableENI {
+				nodes, err := ciliumK8sClient.CiliumV2().CiliumNodes().List(meta_v1.ListOptions{})
```

Comment:
Consider using the `continue` field in the ListOptions to avoid memory spikes.

Reply:
Or, since we are doing `gets` it's probably better to have a cilium node's watcher instead.

Reply:
We only need it once so I'm not sure it's worth to maintain a cache. This is not in the hot path , it's a one-off operation when the operator starts and we have one operator. Not sure it's worth optimizing this much unless it becomes a problem somehow.

Reply:
But who, what is doing the get of a cilium node when the update / updatestatus fail?

Reply:
OK, now I understand. The watcher already exists. What I don't understand is how we can know whether a notification is coming in and when. We can just take what is in the cache and run it with, it might still be outdated. Even if we know that it is outdated, we don't really know whether a notification is still outstanding or whether the cache is corrupt somehow.

I'm not opposed to go into this path but I would prefer to do it on top of this as this resolves a critical bottleneck already.